assets/js/base/context/add-to-cart-form/form/submit/index.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import triggerFetch from '@wordpress/api-fetch';
+import {
+	useAddToCartFormContext,
+	useValidationContext,
+} from '@woocommerce/base-context';
+import { useEffect, useCallback, useState } from '@wordpress/element';
+import { useStoreCart, useStoreNotices } from '@woocommerce/base-hooks';
+import { decodeEntities } from '@wordpress/html-entities';
+
+/**
+ * FormSubmit.
+ *
+ * Subscribes to add to cart form context and triggers processing via the API.
+ */
+const FormSubmit = () => {
+	const {
+		dispatchActions,
+		product,
+		quantity,
+		eventRegistration,
+		hasError,
+		isProcessing,
+		requestParams,
+	} = useAddToCartFormContext();
+	const {
+		hasValidationErrors,
+		showAllValidationErrors,
+	} = useValidationContext();
+	const { addErrorNotice, removeNotice } = useStoreNotices();
+	const { receiveCart } = useStoreCart();
```

Comment:
This would be something that would be good to lift up to the `AddToCartFormContextProvider` (anything Cart specific). We could do this kind of state lifts (I think I have one other comment mentioning this) in follow-ups. This is also an area where there may be shared functionality used by the cart provider (another reason to evaluate/work on separately)

Reply:
I think once we work on form-elements, what needs moving to context will become clearer, so I'm happy to have this in a follow up.